### PR TITLE
Enable offer lookup service to retreave offer information by ID

### PR DESCRIPTION
**Pull request by sven1103**
The restructuring tries to adhere to this component structure:

image

### CODE REVIEW

File: finances/src/main/java/life/qbic/projectmanagement/domain/finances/offer/OfferId.java [hunk 1]
```diff
@@ -1,9 +1,10 @@
-package life.qbic.projectmanagement.finances.offer;
+package life.qbic.projectmanagement.domain.finances.offer;
 
 import java.util.Objects;
+import javax.persistence.AttributeConverter;
 
 /**
- * <b>Offer Id</b>
+ * <b>OfferPreview Id</b>
```
Comment (jenniferboedker) [inline finances/src/main/java/life/qbic/projectmanagement/domain/finances/offer/OfferId.java:7]: Why OfferPreview? Do you want to differentiate between the preview offer ID and the offer ID? Why?

File: finances/src/main/java/life/qbic/projectmanagement/domain/finances/offer/OfferPreview.java [hunk 2]
```diff
@@ -9,15 +9,15 @@
...
+ * <b>OfferPreview</b>
  * <p>
  * Describes an offer object in the context of project management.
  *
  * @since 1.0.0
  */
 @Entity
 @Table(name = "offers")
-public class Offer {
+public class OfferPreview {
```
Comment (jenniferboedker) [inline finances/src/main/java/life/qbic/projectmanagement/domain/finances/offer/OfferPreview.java:20]: Here I think the Preview Context makes sense :+1:

File: finances/src/main/java/life/qbic/projectmanagement/domain/finances/offer/OfferId.java (hunk 1 above)
Comment (sven1103) [reply finances/src/main/java/life/qbic/projectmanagement/domain/finances/offer/OfferId.java:7]: its a search and replace side-effect. Again ...

Comment (sven1103) [reply finances/src/main/java/life/qbic/projectmanagement/domain/finances/offer/OfferId.java]: i adjusted it

**Comment by jenniferboedker** [comment]
@sven1103 the build is broken

**Comment by sven1103** [comment]
> @sven1103 the build is broken

yup, on it

**Review by jenniferboedker — APPROVED** [review]
Nice work. Also makes a lot of sense to me to sort it how you did here 🎉

**Pull request merged**